sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/splittabledofn/SplitResult.java

```diff
@@ -28,7 +28,7 @@
 public abstract class SplitResult<RestrictionT> {
   /** Returns a {@link SplitResult} for the specified primary and residual restrictions. */
   public static <RestrictionT> SplitResult<RestrictionT> of(
-      RestrictionT primary, RestrictionT residual) {
+      @Nullable RestrictionT primary, @Nullable RestrictionT residual) {
```

Comment:
Would the `@Nullabe` annotations below(L36 and L39) be enough?

Reply:
Actually no, because the call `SplitRestriction.of(null, ...)` is reported as bug by tools such as SonarCloud. That is due to implicit `@Nonnull` added by the checkerframework.

Reply:
I see. Thanks for the explanation!